src/adapter/graphql/src/queries/accounts/account.rs

```diff
@@ -41,69 +45,86 @@ impl Account {
     }
 
     #[graphql(skip)]
-    pub(crate) fn from_account_info(account_info: kamu_core::auth::AccountInfo) -> Self {
+    pub(crate) fn from_account(account: kamu_accounts::Account) -> Self {
         Self {
-            account_id: AccountID::from(account_info.account_id.clone()),
-            account_name: account_info.account_name.clone().into(),
-            full_account_info: OnceCell::new_with(Some(account_info)),
+            account_id: AccountID::from(account.id.clone()),
+            account_name: account.account_name.clone().into(),
+            full_account_info: OnceCell::new_with(Some(account)),
         }
     }
 
     #[graphql(skip)]
-    pub(crate) fn from_account_name(name: odf::AccountName) -> Self {
-        Self::new(AccountID::from(odf::FAKE_ACCOUNT_ID), name.into())
+    pub(crate) async fn from_account_id(
+        ctx: &Context<'_>,
+        account_id: odf::AccountID,
+    ) -> Result<Self, InternalError> {
+        let authentication_service =
+            from_catalog::<dyn kamu_accounts::AuthenticationService>(ctx).unwrap();
+
+        let account_name = authentication_service
+            .find_account_name_by_id(&account_id)
+            .await?
+            .expect("Account must exist");
+
+        Ok(Self::new(account_id.into(), account_name.into()))
+    }
+
+    #[graphql(skip)]
+    pub(crate) async fn from_account_name(
+        ctx: &Context<'_>,
+        account_name: odf::AccountName,
+    ) -> Result<Self, InternalError> {
+        let authentication_service = from_catalog::<dyn AuthenticationService>(ctx).unwrap();
+
+        let (account_id, account_name) = authentication_service
+            .find_account_by_name(&account_name)
+            .await?
+            .expect("Account must exist");
+
+        Ok(Self::new(account_id.into(), account_name.into()))
     }
 
     #[graphql(skip)]
-    pub(crate) fn from_dataset_alias(ctx: &Context<'_>, alias: &odf::DatasetAlias) -> Self {
+    pub(crate) async fn from_dataset_alias(
```

Comment:
I feel like we should rethink this function. It seems dangerous to me...

I think on GQL level we shouldn't be having some "default" accounts.

Reply:
Similarly, suggest a different way to resolve single-tenant dataset aliases.

Reply:
yep, it's the same problem - will resolve this and let's discuss in the thread above